runtimes/runtimes/operational-telemetry/aws-spans-exporter.ts

```diff
@@ -0,0 +1,121 @@
+                    resultCallback({ code: ExportResultCode.FAILED })
+                })
+        } catch (error) {
+            diag.error('Failed to extract operational data from spans:', error)
+            resultCallback({ code: ExportResultCode.FAILED })
+        }
+    }
+
+    forceFlush(): Promise<void> {
+        if (this.isShutdown) {
+            diag.warn('Force flush attempted on shutdown exporter')
+        }
+        return Promise.resolve()
+    }
+
+    shutdown(): Promise<void> {
+        if (this.isShutdown) {
+            diag.warn('Duplicate shutdown attempt - exporter is already in shutdown state')
+        }
+        this.isShutdown = true
+        return Promise.resolve()
+    }
+
+    private extractOperationalData(spans: ReadableSpan[]): OperationalTelemetrySchema {
+        const scopeRecords = spans.reduce(
+            (acc, span) => {
+                const scopeName = span.instrumentationLibrary.name
+                if (!acc[scopeName]) {
+                    acc[scopeName] = []
+                }
+                acc[scopeName].push(this.spanToOperationalEvent(span))
+                return acc
+            },
+            {} as Record<string, (CaughtErrorEvent | ServerCrashEvent)[]>
+        )
+        const scopes = Object.entries(scopeRecords).map(([scopeName, scopeSpans]) => {
+            return { scopeName: scopeName, data: scopeSpans }
+        })
+
+        return {
+            sessionId: spans[0].resource.attributes['sessionId'] as string,
+            batchTimestamp: Date.now(),
+            server: {
+                name: spans[0].resource.attributes['service.name'] as string,
+                version: spans[0].resource.attributes['service.version'] as string | undefined,
+            },
+            clientInfo: {
+                name: this.telemetryService.getCustomAttributes()['clientInfo.name'] as string | undefined,
+                extension: {
+                    name: this.telemetryService.getCustomAttributes()['clientInfo.extension.name'] as
+                        | string
+                        | undefined,
+                    version: this.telemetryService.getCustomAttributes()['clientInfo.extension.version'] as
+                        | string
+                        | undefined,
+                },
+                clientId: this.telemetryService.getCustomAttributes()['clientInfo.clientId'] as string | undefined,
+            },
+            scopes: scopes,
+        }
+    }
+
+    private spanToOperationalEvent(span: ReadableSpan): CaughtErrorEvent | ServerCrashEvent {
+        const unixEpochSeconds = span.endTime[0]
+        if (span.name === 'CaughtErrorEvent') {
+            return {
+                name: span.name,
+                timestamp: unixEpochSeconds,
+                errorType: span.attributes['errorType'] as string,
+            } as CaughtErrorEvent
+        }
+        if (span.name === 'ServerCrashEvent') {
```

Comment:
Can we avoid extending the exporter logic every time new metric type is added? It's not scaling, especially when we need it on server side.